_pytest/assertion/truncate.py

```diff
@@ -0,0 +1,102 @@
+"""
+Utilities for truncating assertion output.
+
+Current default behaviour is to truncate assertion explanations at
+~8 terminal lines, unless running in "-vv" mode or running on CI.
+"""
+
+import os
+
+import py
+
+
+DEFAULT_MAX_LINES = 8
+DEFAULT_MAX_CHARS = 8 * 80
+USAGE_MSG = "use '-vv' to show"
+
+
+def truncate_if_required(explanation, item, max_length=None):
+    """
+    Truncate this assertion explanation if the given test item is eligible.
+    """
+    if _should_truncate_item(item):
+        return _truncate_explanation(explanation)
+    return explanation
+
+
+def _should_truncate_item(item):
+    """
+    Whether or not this test item is eligible for truncation.
+    """
+    verbose = item.config.option.verbose
+    return verbose < 2 and not _running_on_ci()
+
+
+def _running_on_ci():
```

Comment:
This is addressed in the original commit message, it attempts to provide some utility for the user: 

     Always show full comparison output if on CI.

    When you don't get enough information with a test running on a CI, it's quite
    frustrating, for various reasons:

    - It's more likely to be a flaky test, so you might not be able to reproduce
      the failure.
    - Passing -vv is quite bothersome (creating a temporary commit and reverting
      it)

    For those reasons, if something goes wrong on CI, it's good to have as much
    information as possible.

Hey @The-Compiler, what makes passing `-vv` bothersome? Is it because it makes the output (other than your failing test) too verbose? I'm not sure I understand why you need a temporary commit. Thanks!